app/controllers/application_controller.rb

```diff
@@ -5,6 +5,123 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "password_security"
   end
 
-end
\ No newline at end of file
+  get "/" do
+    erb :index
+  end
+
+  get "/signup" do
+    redirect '/tweets' if logged_in?
+    erb :'users/create_user'
+  end
+
+  post "/signup" do
+      @user = User.new(:username => params[:username], :email => params[:email], :password => params[:password])
+
+      if @user.save && params[:username].size > 0 && params[:email].size > 0
+        session[:user_id] = @user.id
+        redirect "/tweets"
+      else
+        redirect "/signup?failed=yes"
+      end
+  end
+
+  get '/login' do
+    redirect '/tweets' if logged_in?
+    erb :'users/login'
+  end
+
+  post '/login' do
+
+    if @user = User.find_by(:username => params[:username])
+
+      if !!@user && !!@user.authenticate(params[:password])
+        session[:user_id] = @user.id
+        redirect "/tweets"
+      else
+        redirect '/login?failed=yes'
+      end
+    else
+      redirect '/login?failed=yes'
+    end
+
+  end
+
+  get '/tweets' do
+    logged_in? ? (erb :'tweets/tweets') : (redirect '/login?failed=yes')
+  end
+
+  post '/tweets' do
+    if !params[:content].empty?
```

Comment:
I think you could use [ActiveRecord validations](http://guides.rubyonrails.org/active_record_validations.html) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.